server/src/main/java/org/elasticsearch/plugins/PluginDescriptor.java

```diff
@@ -161,7 +166,16 @@ public void writeTo(final StreamOutput out) throws IOException {
         out.writeString(name);
         out.writeString(description);
         out.writeString(version);
-        Version.writeVersion(elasticsearchVersion, out);
+        if (out.getTransportVersion().before(TransportVersions.PLUGIN_DESCRIPTOR_STRING_VERSION)) {
+            try {
+                Version.writeVersion(Version.fromString(elasticsearchVersion), out);
+            } catch (IllegalArgumentException e) {
+                // can't parse current version, so send placeholder
+                Version.writeVersion(BWC_PLACEHOLDER_VERSION, out);
```

Comment:
I like this solution; I was contemplating of doing something similar myself (we are solving basically the same problem in 2 places - https://github.com/elastic/elasticsearch/pull/100746 - but I was wondering what version could be used. I was thinking Version.CURRENT. What made you choose a fixed one?
(I will update my PR to be consistent with yours afterwards)

Reply:
I would use Version.current if possible, is there a reason it can't work?

Reply:
Is `Version.CURRENT` going to stick around indefinitely? I went with a placeholder because I wasn't sure if we were going to get rid of `Version.CURRENT` altogether at some point.

Reply:
We do want to get rid of it. Perhaps we can defer what to do here because this still doesn't seem like a case we need to catch/send a fallback, as noted in my other comment on this block.